src/NuProj.Tests/GenerateNuSpecTests.cs

```diff
@@ -197,5 +198,81 @@ public void Task_GenerateNuSpec_UseInputFileName()
+
+            var result = await MSBuild.ExecuteAsync(projectPath, "GenerateNuSpec");
+            result.AssertError("The 'Id' property cannot be empty or undefined.");
+        }
+
+        [Theory]
+        [InlineData("Task_GenerateNuSpec_NoRequiredParameters", @"NuGetPackage\NuGetPackage.nuproj")]
+        public async System.Threading.Tasks.Task Task_GenerateNuSpec_NoRequiredParameters_IdOnly(string scenarioName, string projectToBuild)
+        {
+            var solutionDir = Assets.GetScenarioDirectory(scenarioName);
+            var projectPath = Path.Combine(solutionDir, projectToBuild);
+
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                {"Id", "NoRequiredParameters"}
+            };
+
+            var result = await MSBuild.ExecuteAsync(projectPath, "GenerateNuSpec", properties);
+            result.AssertError("The 'Version' property cannot be empty or undefined.");
+        }
+
+        [Theory]
+        [InlineData("Task_GenerateNuSpec_NoRequiredParameters", @"NuGetPackage\NuGetPackage.nuproj")]
+        public async System.Threading.Tasks.Task Task_GenerateNuSpec_NoRequiredParameters_IdAndVersion(string scenarioName, string projectToBuild)
+        {
+            var solutionDir = Assets.GetScenarioDirectory(scenarioName);
+            var projectPath = Path.Combine(solutionDir, projectToBuild);
+
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                {"Id", "NoRequiredParameters"},
+                {"Version", "1.0.0"}
+            };
+
+            var result = await MSBuild.ExecuteAsync(projectPath, "GenerateNuSpec", properties);
+            result.AssertError("required parameter \"Authors\"");
+            result.AssertError("required parameter \"Description\"");
+        }
+
+        [Theory]
+        [InlineData("Task_GenerateNuSpec_NoRequiredParameters", @"NuGetPackage\NuGetPackage.nuproj", @"NuGetPackage\template.nuspec")]
+        public async System.Threading.Tasks.Task Task_GenerateNuSpec_NoRequiredParameters_FromTemplate(string scenarioName, string projectToBuild, string templateFile)
+        {
+            var solutionDir = Assets.GetScenarioDirectory(scenarioName);
+            var projectPath = Path.Combine(solutionDir, projectToBuild);
+            var templatePath = Path.Combine(solutionDir, templateFile);
+
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                {"NuSpecTemplate", templatePath}
+            };
+
+            // load the template manifest so we can compare it later..
+            Manifest templateManifest;
+            using (var inputStream = File.OpenRead(templatePath))
+                templateManifest = Manifest.ReadFrom(inputStream, false);
+
+            var result = await MSBuild.ExecuteAsync(projectPath, "GenerateNuSpec", properties);
+            result.AssertSuccessfulBuild();
```

Comment:
Also getting `The "GenerateNuSpec" task was not given a value for the required parameter "Id".` here.